istio-2-mixin/variables.libsonnet

```diff
@@ -0,0 +1,204 @@
+            value=true,
+            customAllValue='.+'
+          )
+          + var.query.selectionOptions.withMulti(
+            true,
+          )
+          + var.query.refresh.onTime()
+          + var.query.withSort(
+            i=1,
+            type='alphabetical',
+            asc=true,
+            caseInsensitive=false
+          );
+        std.mapWithIndex(chainVarProto, utils.chainLabels(groupLabels, [])),
+      datasources: {
+        prometheus:
+          var.datasource.new('datasource', 'prometheus')
+          + var.datasource.generalOptions.withLabel('Data source')
+          + var.datasource.withRegex(''),
+        loki:
+          var.datasource.new('loki_datasource', 'loki')
+          + var.datasource.generalOptions.withLabel('Loki data source')
+          + var.datasource.withRegex('')
+          // hide by default (used for annotations)
+          + var.datasource.generalOptions.showOnDashboard.withNothing(),
+      },
+      local createOverviewVariable(name, displayName, metric, selector) =
+        local variable = var.query.new(name)
+        + var.query.withDatasourceFromVariable(root.datasources.prometheus)
+        + var.query.queryTypes.withLabelValues(
+          'pod',
+          '%s{%s}' % [metric, selector],
+        )
+        + var.query.generalOptions.withLabel(displayName)
+        + var.query.selectionOptions.withIncludeAll(
+          value=true,
+        )
+        + var.query.selectionOptions.withMulti(
+          true,
+        )
+        + var.query.refresh.onTime()
+        + var.query.withSort(
+          i=1,
+          type='alphabetical',
+          asc=true,
+          caseInsensitive=false
+        );
+        [variable],
+      local createQueryVariable(name, displayName, query, regex, includeAll) =
+        local variable = var.query.new(name, query)
+        + var.query.generalOptions.withLabel(displayName)
+        + var.query.withDatasourceFromVariable(root.datasources.prometheus)
+        + var.query.withRegex(regex)
+        + var.query.selectionOptions.withIncludeAll(
+          value=if (!includeAll) then false else true,
+        )
+        + var.query.selectionOptions.withMulti(
+          true,
+        )
+        + var.query.refresh.onTime()
+        + var.query.withSort(
+          i=1,
+          type='alphabetical',
+          asc=true,
+          caseInsensitive=false
+        );
+        [variable],
+      overviewVariables:
+        [root.datasources.prometheus]
+        + groupVariablesFromLabels(groupLabels)
+        + createOverviewVariable(istiodLabel, 'Istiod', 'pilot_info', 'job=~"$job", cluster=~"$cluster"')
+        + createOverviewVariable(gatewayLabel, 'Gateway', 'istio_agent_process_cpu_seconds_total', 'job=~"$job", cluster=~"$cluster", pod=~"istio-egress.*|istio-ingress.*"')
+        + createOverviewVariable(proxyLabel, 'Proxy', 'istio_agent_process_cpu_seconds_total', 'job=~"$job", cluster=~"$cluster", pod!~"istio-egress.*|istio-ingress.*"'),
```

Comment:
Before the generated json had a sort variable set to 0 or 5(for "proxy" this may be a mistake from the old dashboard sent in review) for all the datasources. Now they appear to be 1